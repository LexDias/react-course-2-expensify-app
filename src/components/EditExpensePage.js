import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.editExpenseHandler(this.props.expense.id, expense);
        this.props.history.push('/');
    };
    onRemoveClick = () => {
        this.props.startRemoveExpense(this.props.expense.id);
        this.props.history.push('/');
    };
    render() {
        return (
            <div>
                <h1>Edit Expense</h1>
                <ExpenseForm 
                    expense={this.props.expense}
                    onSubmit={this.onSubmit} 
                />
                <button 
                    onClick={this.onRemoveClick}>
                    Remove
                </button>
            </div>
        );
    }
}

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
    editExpenseHandler: (id, expense) => dispatch(editExpense(id, expense)),
    startRemoveExpense: (id) => dispatch(startRemoveExpense(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);