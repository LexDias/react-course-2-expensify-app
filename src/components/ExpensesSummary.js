import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total.js';
import expenses from '../selectors/expenses';

export const ExpenseSummary = (props) => {
    const expenseWord = props.expensesCount === 1 ? 'expense' : 'expenses';
    return (
        <div>
            <h1>Viewing {props.expensesCount} {expenseWord} totalling {numeral(props.expensesTotal / 100).format('$0,0.00')}</h1>
        </div>
    )
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    }
};

export default connect(mapStateToProps)(ExpenseSummary);
