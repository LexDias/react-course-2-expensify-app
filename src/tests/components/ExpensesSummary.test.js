import React from 'react';
import { shallow } from 'enzyme';
// importa o componente nao conectado ao store, para que se possa passar as props
import { ExpenseSummary } from '../../components/ExpensesSummary'; 

test('should render ExpenseSummary with one expense', () => {
    const wrapper = shallow(<ExpenseSummary expensesCount={1} expensesTotal={295} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseList with multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expensesCount={23} expensesTotal={29532424} />);
    expect(wrapper).toMatchSnapshot();
});