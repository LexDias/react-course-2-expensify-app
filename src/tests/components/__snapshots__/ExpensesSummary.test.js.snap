// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $295,324.24
  </h1>
</div>
`;

exports[`should render ExpenseSummary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.95
  </h1>
</div>
`;
